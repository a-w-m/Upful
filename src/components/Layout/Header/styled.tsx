import React from "react"
import styled from "styled-components"
import device from "../mediaQuery"
import { Link } from "gatsby"

export const HeaderContainer = styled.header`
  
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: flex-end;
  padding: 2rem 2rem 0rem 2rem;

  @media ${device.tablet}{
    justify-content: flex-end;
    gap: 2rem;
    grid-area: header;

    /*remove logo from header at tablet width and larger */
    & > a:nth-child(2){
      display: none;
    }

  }

  @media ${device.laptop}{
    padding: 2rem 4rem 0rem 4rem;
  }
`

export const HamburgerNavContainer = styled.div<{open: boolean}>`
width: ${({open})=> open? "100%": "auto"};
align-self: flex-start;
@media ${device.laptop}{
  order: 1;
  flex-basis: 100%;
}

`

export const LogoContainer = styled.div`
  max-width: 960;
  flex-basis: 100%;
  display: flex;
  justify-content: center;
`

export const LogoWrapper = styled.div`
  flex-basis: 25%;

  @media ${device.laptop} {
    flex-basis: 15%;
  }
`

export const LinkWrapper = styled.h1`
  margin: 0;
  font-family: var(--main-font);
`

export const TitleLink = styled(props => <Link {...props} />)`
  color: #000;
  text-decoration: none;
`

export const CartContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  align-self: flex-start;
`
export const Checkout = styled.button`
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  line-height: 1.5;
  text-decoration: none;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  flex-direction: row;
  -webkit-box-align: center;
  align-items: center;
  font-size: 1rem;
  color: #666;
  font-family: Open Sans, sans-serif;
  font-family: var(--fontFamily);
  font-weight: 700;
`

export const Count = styled.span`
  align-self: flex-end;
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 2rem;
`
