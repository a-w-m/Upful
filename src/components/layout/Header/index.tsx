import PropTypes from "prop-types"
import React from "react"
const { useState, useRef, useContext } = React

import Hamburger from "../Hamburger"
import Nav from "../Nav"
import { Link } from "gatsby"

import { StaticImage } from "gatsby-plugin-image"
import { SnipcartContext } from "gatsby-plugin-snipcart-advanced/context.js"

import {
  HeaderContainer,
  LogoContainer,
  LogoWrapper,
  CartContainer,
  Checkout,
  Count,
  HamburgerNavContainer
  
} from "./styled"
import { Cart } from "../../Icons"
import useOnClickOutside from "../../../hooks/useOnClickOutside"

interface HeaderProps {
  siteTitle: String
}

const Header: React.FC<HeaderProps> = ({ siteTitle }) => {
  const node = useRef(null)
  const [open, setOpen] = useState(false)
  useOnClickOutside(node, () => setOpen(false))
  const { state } = useContext(SnipcartContext)
  const { cartQuantity } = state

  return (
    <HeaderContainer>
      <HamburgerNavContainer ref={node} open ={open}>
        <Hamburger open={open} setOpen={setOpen} />
        <Nav open={open} />
      </HamburgerNavContainer>
      <CartContainer>
        <Checkout className="snipcart-checkout">
          <Cart />
        </Checkout>
        <Count>{cartQuantity}</Count>
      </CartContainer>
      <LogoContainer>
        <LogoWrapper>
          <Link to="/">
            <StaticImage
              src="../../../images/upful-gold-frame-logo.png"
              alt="logo"
              style={{ borderRadius: "100%" }}
              width={250}
            />
          </Link>
        </LogoWrapper>
      </LogoContainer>
    </HeaderContainer>
  )
}

Header.propTypes = {
  siteTitle: PropTypes.string.isRequired,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
