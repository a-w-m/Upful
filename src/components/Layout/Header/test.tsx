import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import Header from ".."

describe("Header component", () => {
  test("matches snapshot", () => {
    const { container } = render(<Header children="" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("renders Header with correct site title", () => {
    render(<Header children="" />)

    expect(screen.getByText("StoreFront")).toBeInTheDocument()
  })

  // test("displays toggle menu on small screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.getByAltText("toggle-menu")).toBeInTheDocument()

  // })

  // test("does not display toggle menu on large screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.queryAllByAltText("toggle-menu")).toBeNull()

  // })
})

// describe("click outside toggle menu", () => {
//   test("Nav remains visible when clicking inside component ", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.click(Nav)
//     expect(Nav).toBeVisible()
//   })

//   test("Nav is not visible when clicking outside component", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.mouseDown(Button)
//     expect(Nav).not.toBeVisible()
//   })
// })
