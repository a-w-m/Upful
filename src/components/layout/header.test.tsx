import React from "react"
import { render, screen } from "@testing-library/react"
import Header from "./layout"

describe("Header component", () => {
  test("matches snapshot", () => {
    const { container } = render(<Header children="" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("renders Header with correct site title", () => {
    render(<Header children="" />)

    expect(screen.getByText("StoreFront")).toBeInTheDocument()
  })
})
