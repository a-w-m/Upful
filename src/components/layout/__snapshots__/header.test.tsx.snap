// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header
  class="sc-dlfnbm mFHZk"
>
  <div>
    <button
      aria-controls="link-list"
      aria-expanded="false"
      aria-label="open navigation"
      class="sc-bdfBwQ fwZoWG"
      type="button"
    >
      <div />
      <div />
      <div />
    </button>
    <nav
      class="sc-gsTCUz EzLyR"
    >
      <a
        href="/"
      >
        About Us
      </a>
      <a
        href="/"
      >
        Clothing
      </a>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/"
      >
        Accessories
      </a>
    </nav>
  </div>
  <div
    class="sc-hKgILt fEkVZL"
  >
    <h1
      class="sc-eCssSg iHXJeJ"
    >
      <a
        class="sc-jSgupP ZBvTP"
        href="/"
      >
        StoreFront
      </a>
    </h1>
  </div>
</header>
`;
